### PR TITLE
Cat 2486 refacturing drone, adding jump high and jump long bricks for JumpingSumo drone

### PR DESCRIPTION
**Pull request by cpozvek**

### CODE REVIEW

File: catroid/src/main/java/org/catrobat/catroid/content/bricks/JumpingSumoBasicBrick.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+		return prototypeView;
+	}
+
+	@Override
+	public int getRequiredResources() {
+		return super.getRequiredResources() | Brick.JUMPING_SUMO;
+	}
+
+	protected abstract String getBrickLabel(View view);
```
Comment (thmq) [inline catroid/src/main/java/org/catrobat/catroid/content/bricks/JumpingSumoBasicBrick.java]: The changes I suggested above should most likely leave this method obsolete. What I'm trying to say is: please remove everything you don't absolutely need.

File: catroid/src/main/java/org/catrobat/catroid/drone/jumpingsumo/JumpingSumoBrickFactory.java [hunk 2]
```diff
@@ -23,18 +23,25 @@
 package org.catrobat.catroid.drone.jumpingsumo;
 
 import org.catrobat.catroid.content.bricks.BrickBaseType;
+import org.catrobat.catroid.content.bricks.JumpingSumoJumpHighBrick;
+import org.catrobat.catroid.content.bricks.JumpingSumoJumpLongBrick;
 
 public final class JumpingSumoBrickFactory {
```
Comment (thmq) [inline catroid/src/main/java/org/catrobat/catroid/drone/jumpingsumo/JumpingSumoBrickFactory.java]: Do you really need a factory for this?

File: catroid/src/main/java/org/catrobat/catroid/content/actions/JumpingSumoJumpHighAction.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	@Override
+	protected void begin() {
+		super.begin();
+		controller = JumpingSumoDeviceController.getInstance();
+		deviceController = controller.getDeviceController();
+		if (deviceController != null) {
+			deviceController.getFeatureJumpingSumo().sendAnimationsJump(ARCOMMANDS_JUMPINGSUMO_ANIMATIONS_JUMP_TYPE_ENUM.ARCOMMANDS_JUMPINGSUMO_ANIMATIONS_JUMP_TYPE_HIGH);
+			Log.d(TAG, "send jump high command JS down");
+		} else {
+			Log.d(TAG, "error: send jump high command JS");
```
[the remark this answers is not in the document]
Comment (cpozvek) [reply catroid/src/main/java/org/catrobat/catroid/content/actions/JumpingSumoJumpHighAction.java]: That was for debugging. I will delete it

File: catroid/src/main/java/org/catrobat/catroid/content/actions/JumpingSumoJumpLongAction.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+	@Override
+	protected void begin() {
+		super.begin();
+		controller = JumpingSumoDeviceController.getInstance();
+		deviceController = controller.getDeviceController();
+		if (deviceController != null) {
+			deviceController.getFeatureJumpingSumo().sendAnimationsJump(ARCOMMANDS_JUMPINGSUMO_ANIMATIONS_JUMP_TYPE_ENUM.ARCOMMANDS_JUMPINGSUMO_ANIMATIONS_JUMP_TYPE_LONG);
+			Log.d(TAG, "send jump long command JS down");
+		} else {
```
[the remark this answers is not in the document]
Comment (cpozvek) [reply catroid/src/main/java/org/catrobat/catroid/content/actions/JumpingSumoJumpLongAction.java]: No

File: catroid/src/main/java/org/catrobat/catroid/content/bricks/JumpingSumoBasicBrick.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+import org.catrobat.catroid.R;
+
+public abstract class JumpingSumoBasicBrick extends BrickBaseType {
+
+	@Override
+	public View getView(Context context, int brickId, BaseAdapter baseAdapter) {
+		if (animationState) {
+			return view;
+		}
+		if (view == null) {
```
[the remark this answers is not in the document]
Comment (cpozvek) [reply catroid/src/main/java/org/catrobat/catroid/content/bricks/JumpingSumoBasicBrick.java]: Ok will do that

File: catroid/src/main/java/org/catrobat/catroid/drone/jumpingsumo/JumpingSumoBrickFactory.java (hunk 2 above)
Comment (cpozvek) [reply catroid/src/main/java/org/catrobat/catroid/drone/jumpingsumo/JumpingSumoBrickFactory.java]: We use this for our Defaultproject

**Comment by cheidenreich** [comment]
please run espresso tests

**Comment by cheidenreich** [comment]
please run espresso tests

**Comment by cpozvek** [comment]
Up-to-date, can be merged

**Pull request merged**